torchtune/recipe_interfaces.py

```diff
@@ -58,3 +58,36 @@ def cleanup(self, **kwargs) -> None:
         Any cleaning up needed for the recipe.
         """
         ...
+
+
+class EvalRecipeInterface(Protocol):
+    """
+    This class provides a loose structure which every LLM evaluation recipe
+    should follow. Please note that the interface itself should not be a vehicle for
+    code reuse. TorchTune strictly prohibits implementation inheritance in the codebase.
+    """
+
+    def load_checkpoint(self, **kwargs) -> None:
```

Comment:
why do you need an explicit `load_checkpoint` if this is just going to call `checkpointer.load_checkpoint()` in most cases?

Reply:
Following what I see in other recipes, good to be consistent. Do you think it's overkill?

Reply:
The recipe ```load_checkpoint``` contains any logic that is needed after the state_dict is returned from the checkpoint. So for example if you want to extract the adapter weights separately and do stuff with it.